Source/JavaScriptCore/jit/OperationResult.h

```diff
@@ -34,18 +34,21 @@
 
 namespace JSC {
 
+template<typename T>
+constexpr bool assert_not_operation_signature =
+    []{ static_assert(!std::is_function_v<typename std::remove_pointer_t<T>>, "This should take a return type, not an operator signature."); return true; }();
```

Comment:
Why does this need to be wrapped in a lambda?

Reply:
It needs to be an expression, static_assert is a statement. This makes sure that SFINAE can't just ignore it, which is what the previous code was doing.

Reply:
Before, when you did operationExceptionRegister<operationBlah>, it would silently give you the non-void version

Reply:
Can we change this to:
```
// This is wrapped in a lambda since it's used by static_asserts thus needs to be an expression.
template<typename T>
constexpr bool assertNotOperationSignature = ([] { 
    static_assert(!std::is_function_v<typename std::remove_pointer_t<T>>, "This should take a return type, not an operator signature."); 
    return true; 
})();
```